src/vos/vos_io.c

```diff
@@ -2429,26 +2434,27 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err,
 	if (err != 0)
 		goto abort;
 
-	/* The object is already held for md-on-ssd phase2 pool */
-	if (ioc->ic_obj == NULL) {
```

Comment:
why removed the check "if (ioc->ic_obj == NULL)"

Reply:
I used a separate "ic_pinned_obj" now.